### ISSUE TITLE
Allow arguments for commands in custom keyboard shortcuts

### ISSUE THREAD

**Issue by kjk** [sumatrapdfreader/sumatrapdf#4346]
Some commands could use arguments. For example `CmdCreateAnnotHighlight` could use a color argument so that we could do:
```
Shortcuts [
	[
		Cmd = CmdCreateAnnotHighlight #00ff00   
		Key = b
	]
]
```
This would address requests like https://github.com/sumatrapdfreader/sumatrapdf/discussions/4344, https://github.com/sumatrapdfreader/sumatrapdf/issues/4338, https://github.com/sumatrapdfreader/sumatrapdf/issues/3607 and possibly more

This requires 2 things:
* [x] implement a generic system that allows arguments with commands
* [ ] identify commands that should accept arguments and change their implementation to use the arguments

**Comment by kjk** [comment]
Allow color arguments for the following commands:
```
        case CmdCreateAnnotText:
        case CmdCreateAnnotLink:
        case CmdCreateAnnotFreeText:
        case CmdCreateAnnotLine:
        case CmdCreateAnnotSquare:
        case CmdCreateAnnotCircle:
        case CmdCreateAnnotPolygon:
        case CmdCreateAnnotPolyLine:
        case CmdCreateAnnotHighlight:
        case CmdCreateAnnotUnderline:
        case CmdCreateAnnotSquiggly:
        case CmdCreateAnnotStrikeOut:
        case CmdCreateAnnotRedact:
        case CmdCreateAnnotStamp:
        case CmdCreateAnnotCaret:
        case CmdCreateAnnotInk:
        case CmdCreateAnnotPopup:
        case CmdCreateAnnotFileAttachment: {
```

**Comment by GitHubRulesOK** [comment]
Wow 
can we have something similar  for call external app 0-9 with %filename %page and %selection 
to replace external viewers and more ?

**Comment by H5820121** [comment]
@GitHubRulesOK
It would be very helpful if it was possible (in the advanced settings) to
set a keyboard shortcut for external viewers 0-9!


‫בתאריך שבת, 29 ביוני 2024 ב-20:41 מאת ‪GitHubRulesOK‬‏ <‪
***@***.***‬‏>:‬

**Comment by kjk** [comment]
Sure, show me how you would like it to work.

**Comment by GitHubRulesOK** [comment]
Unsure on detailed concept but currently too run the external view commands it is 3 keys 
so for example we need ALT & F & A but if it was either numeric or main key board addressable 1-9 based it would be easier
some duplication might be reduced so 
CTRL + C =copy to clipboard but CTRL+C reassigned to 
**poor example** external `cmd.exe programable-notepad %f %p %s` allows for simpler usage of clip

**Comment by H5820121** [comment]
@Krzysztof Kowalczyk
Shortcuts [
[
Cmd = CmdExternalViewers 0   (This refers to the first on the
ExternalViewers list)
Key =
]
 [
Cmd = CmdExternalViewers 1   (This refers to the second in the
ExternalViewers list)
Key =
]

]


‫בתאריך שבת, 29 ביוני 2024 ב-21:32 מאת ‪GitHubRulesOK‬‏ <‪
***@***.***‬‏>:‬

**Comment by kjk** [comment]
I could add `CmdExternalCommand` e.g.
```
[
   Cmd = CmdExternalCommand acrobat.exe ...args
   Key = 
]
[
   Cmd = CmdExternalCommand https://translate.google.com/?op=translate&sl=auto&tl=${userlang}&text=${selection}
   Key = 
]
```

which could kind of subsume `ExternalViewers` and `SelectionHandlers` except they don't require a shortcut assignment so it's weird to put them in shortcuts.

A possible solution would be to add a way to define new commands, like:
```
Commands [
  [
    Name = CmdTranslateSelectionWithGoogle
    Cmd = CmdExternalCommand https://translate.google.com/?op=translate&sl=auto&tl=${userlang}&text=${selection}
    Type = SelectionHandler
    Key = 
  ]
]
```
Type would be `SelectionHandler` or `ExternalCommand` (or `ExternalViewer for compatibility with current terminology).

`Key` is optional because it could also be assigned via
```
Shortcuts [
  [
    Cmd = CmdTranslateSelectionWithGoogle
    Key = Ctrl + b
  ]
]
```

The I could upgrade settings file and convert `SelectionHandlers` and `ExternalViewers` to `Commands` section and deprecate them.

**Comment by H5820121** [comment]
@kjk <https://github.com/kjk>
@GitHubRulesOK
"Closed #4346 as completed"
Was there anything done in pre-release about setting hotkeys to external
viewers?


‫בתאריך יום א׳, 30 ביוני 2024 ב-0:20 מאת ‪Krzysztof Kowalczyk‬‏ <‪
***@***.***‬‏>:‬
